transmogrifier/sources/xml/datacite.py

```diff
@@ -302,20 +246,117 @@ def get_optional_fields(self, xml: Tag) -> dict | None:
         return fields
 
     @classmethod
-    def get_main_titles(cls, xml: Tag) -> list[str]:
+    def get_alternate_titles(
+        cls, source_record: Tag
+    ) -> list[timdex.AlternateTitle] | None:
+        alternate_titles = []
+        alternate_titles.extend(
+            [
+                timdex.AlternateTitle(
+                    value=str(title.string.strip()),
+                    kind=title["titleType"],
+                )
+                for title in source_record.find_all("title", string=True)
+                if title.get("titleType")
+            ]
+        )
+        alternate_titles.extend(list(cls._get_additional_titles(source_record)))
+        return alternate_titles or None
+
+    @classmethod
+    def _get_additional_titles(
+        cls, source_record: Tag
+    ) -> Iterator[timdex.AlternateTitle]:
+        """Get additional titles from get_main_titles method."""
+        for index, title in enumerate(cls.get_main_titles(source_record)):
+            if index > 0:
+                yield timdex.AlternateTitle(value=title)
```

Comment:
Hmm, the creation of the private method `_get_additional_titles()` brings the number of `title`-related methods to 4: 
* `get_valid_title`
* `get_main_titles`
* `get_alternate_titles`
* `_get_additional_titles`

Because of this, I personally prefer moving the logic in `_get_additional_titles()` private method into `get_alternate_titles()` and adding a comment instead. If not, maybe rename to `_get_additional_alternate_titles()`. 

That said, I think we did kinda' discuss that `get_valid_title()` might be revisited at some point...

Curious what @ghukill thinks!

Reply:
Thanks for raising @jonavellecuerdo.  Without comment on your proposal above, I see something else in these title parsings that also stood out, after some hopping back and forth.

Looks like maybe the same XML parsing is used [here in get_main_titles()](https://github.com/MITLibraries/transmogrifier/blob/bdb2c4bd95c930683f7a5b6c2499fc88fe03eebb/transmogrifier/sources/xml/datacite.py#L357-L359) and then again [here in get_alternate_titles()](https://github.com/MITLibraries/transmogrifier/blob/bdb2c4bd95c930683f7a5b6c2499fc88fe03eebb/transmogrifier/sources/xml/datacite.py#L255-L260)?  The difference being the first is looking for strings, and the second is returning `timdex.AlternateTitle` instances.

I don't know what the solution is, something is feeling a bit wonky here.  

Wouldn't this result in duplicating values for "main titles" and "alternate titles"?  Does that have some bearing here?

Reply:
This is another one where I agree 100% but I think we're better off creating an issue to address this separately. This is getting too close to core logic that I'm hesitant to mess with it during the field method refactor. Agree or disagree with that approach?

Reply:
@ghukill I think the difference between `get_main_titles()` and `get_alternate_titles` is:
* `get_main_titles()` retrieves values where attribute `@titleType` is None. 
* `get_alternate_titles()` retrieves values where attribute `@titleType` exists.
* `_get_additional_titles` retrieves all values from `get_main_titles()`, excluding the first item in the list (if there are multiple values returned for `get_main_titles()`, the first value is used for `timdexRecord.title`.

That said, @ehanson8 you make a good point! This is stepping into the logic as opposed to the field method. I started an [issue for Datacite transform logic](https://github.com/MITLibraries/transmogrifier/issues/179).

Reply:
Agreed @ehanson8, and thanks for the eagle eyes @jonavellecuerdo!  This is like the 100th time this week I've missed a `not` in code.  I might need new glasses. 

Thanks for starting the issue.  It does feel like something that could get untangled and streamlined, but agree with all comments above.

Reply:
Thanks again for creating the issue, though I would argue that `title` issue goes much deeper than just `Datacite` so you could frame it more broadly as `Refactor title logic`. I think other transforms may be equally messy re: titles